### PR TITLE
support passing encoded image

### PR DESCRIPTION
**Pull request by Achoobert**
Reject on file being too big

https://github.com/digi-serve/ab_service_file_processor/issues/36
Needed to get this file added
https://github.com/digi-serve/ab_service_file_processor/pull/37

### CODE REVIEW

File: api/controllers/file_processor/file-base64-upload.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+   // Package the Find Request and pass it off to the service
+   req.ab.log(`file_processor::file-base64-upload`);
+
+   // verify your inputs are correct:
+   if (
+      !(req.ab.validUser(/* false */)) ||
+      !req.ab.validateParameters(inputParams /*, false , valuesToCheck*/)
+   ) {
+      req.ab.log("... base64Params invalid");
+      req.ab.log(error);
```
Comment (nh758) [inline api/controllers/file_processor/file-base64-upload.js:43]: `error` here refers to `console.error` defined on line 20? So we're logging the function?
Also `validateParameters` should have its own error handling. Not sure logging here is necessary.

File: README.md [hunk 2]
```diff
@@ -40,6 +40,7 @@ ___
 - [File](#File)
   - [Get a File](#Get-a-File)
   - [Get a File as base64](#Get-a-File-as-base64)
+  - [Upload a File as base64](#Upload-a-File-as-base64)
```
Comment (nh758) [inline README.md:43]: Did you use `npm run doc`? It should have generated more than this.

File: api/controllers/file_processor/file-base64-upload.js [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+ * @apiSuccess (200) {Object} data
+ * @apiSuccess (200) {string} data.uuid
+ */
+
+const async = require("async");
+const { error } = require("console");
+
+var inputParams = {
+   file: { string: true, required: true },
+   photoID: { string: true, required: true },
```
Comment (nh758) [inline api/controllers/file_processor/file-base64-upload.js:24]: Change to `fileID` to be consistent? Not necessarily a photo.

File: api/controllers/file_processor/file-base64-upload.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+         (next) => {
+            var jobData = {
+               photoID: photoID,
+               object: objID,
+               field: fieldID,
+               file: file, // raw base64 file
+               size: size,
+               type: type,
+               fileName: fileName,
+               uploadedBy: uploadedBy || req.ab.userDefaults().username,
```
Comment (nh758) [inline api/controllers/file_processor/file-base64-upload.js:108]: `uploadedBy` is a required param in the validation rules. So shouldn't need a fallback.

File: api/controllers/file_processor/file-base64-upload.js (hunk 3 above)
Comment (nh758) [reply api/controllers/file_processor/file-base64-upload.js:24]: Also should this be optional? Normally we don't expect the client to assign the `uuid`. Making this optional would make the route more flexible for other use cases.

File: api/controllers/file_processor/file-base64-upload.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+const async = require("async");
+const { error } = require("console");
+
+var inputParams = {
+   file: { string: true, required: true },
+   photoID: { string: true, required: true },
+   objID: { string: true, required: true },
+   fieldID: { string: true, required: true },
+   uploadedBy: { string: true, required: true },
+   size: { number: true, required: true },
```
Comment (nh758) [inline api/controllers/file_processor/file-base64-upload.js:28]: Do we trust the client to provide an accurate size? It seems like we calculate it anyways on line 87?

**Review by nh758 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Achoobert) [reply README.md:43]: Good shout, didn't know about that command. Running it right now seems to tag my current branch name, but it did show me what params I needed to document better

File: api/controllers/file_processor/file-base64-upload.js (hunk 5 above)
Comment (Achoobert) [reply api/controllers/file_processor/file-base64-upload.js:28]: Was mostly copying params from old methods: a few of these may be trusting the client too much, but this one _is_ getting calculated anyway.

**Review by nh758 — APPROVED** [review]

**Review by wongpratan — APPROVED** [review]

**Pull request merged**